reputation/microeconomic_reputation.tex

```diff
@@ -0,0 +1,367 @@
+In an usual case the "visible decision" will by caused by the algorithm implemented in the provider/requestor agent,
+but this is just an effect of the agent's "real decision" about the implementation.\footnote{In an extreme case we can imagine 
+a human operator who directly interacts with the Golem protocol. Or just imagine someone who turns off the currently rented hardware.}
+
+The agent's core decision-making process is called a "strategy". The better agent fulfills their part of the agreements, the more "honest" is the strategy.
+So, in other words: when agent $A$ signs an agreement $a$ with an agent using a dishonest strategy, the expected value of $V_{AL}(a)$ is high.
+When the other side is fully honest, $V_{AL}(a) = 0$. 
+Note that agent's intentions don't matter here - there's no difference if an agent breaks an agreement on purpose or accidentally.
+ 
+Agent always tries to maximize their own utility (the "expected utility maximization" assumption), in other words: agent always selects the most profitable strategy.
+Let's now paraphrase the third goal from the previous section as:
+
+\begin{conclusion}
+
+The final purpose of the reputation system on Golem is to make honest strategies more profitable than dishonest strategies.
+
+\label{main purpose conclusion}
+\end{conclusion}
+
+
+\subsection{Non-reputation solutions}
+
+Let's consider few directions towards the goal defined in the previous section:
+
+\begin{itemize}
+\item Make dishonest strategies not available at all, e.g. ensure that debit note acceptance forces payment.
+\item Make dishonest strategies hard to implement, e.g. hide/obfuscate some important components of Golem.
+\item Add some mechanics that directly penalize dishonesty, e.g. require deposits and confiscate them when dishonesty is proven.
+\end{itemize}
+
+It's important to note that - while they have the same purpose as the reputation - they have nothing else to do with anything we call the "reputation".
+
+\begin{conclusion}
+Golem Factory works on the reputation system because we believe it's the best way to achieve goal defined in Conclusion \ref{main purpose conclusion}.
+\end{conclusion}
+
+\subsection{Defining reputation}
+
+$V_{whatever}$ is a determined, known value. When trading on the Golem market, some values are known from the start (e.g. $V_{AN}(a)$), 
+and other only post factum (e.g. $V_{AL}(a)$, so also $V_A(a)$).
+When making decisions under uncertainity, we're using expected values - they will be written as $E(...)$, e.g. $E(V_A(a))$.
+
+Imagine an honest agent $A_1$ who considers signing an agreement $a$ with agent $A_2$. The decision algorithm can be rougly summarized as:
+\begin{enumerate}
+\item Calculate the expected value of the decision \textbf{not} to sign the agreement, $E_{A_1}(\neg a)$
+\item Calculate the expected value of the decision to sign the agreement $E_{A_1}(a) = V_{A_1N}(a) - E(V_{A_1L}(a))$
+\item Sign the agreement if $E_{A_1}(a) > E_{A_1}(\neg a)$
```

Comment:
The elephant in the room seems to be how to calculate this values.

Reply:
For sure!
There are two directions from here:

1. Design a specific reputation system with additional assumptions that make it possible to calculate the values
2. Design a specific reputation system that can be proven to be correct without calculation specific values

I'm not sure if the second path is really possible, but I'd say it might be.

Do you think such discussion should be a part of this document?

Reply:
I daresay, without it the practical applications of this paper seem limited.

But maybe I misunderstood the purpose of this exercise, and practical applications are not what you have in mind.

Reply:
> From my POV, the purpose of this paper is to:
> 
>     * Create a shared definition of the "reputation system"
> 
>     * Agree on a general path towards the reputation system (p. 3.4)
> 
>     * Agree on the elements of the reputation system. E.g. if I understand correctly, you assume something like "fault attribution mechanism" is necessary. I think it is not. We must agree on this to effectively continue works on the reputation.

Forgive me for saying this, but so far none of these goals of the paper seem to have been reached.

> 
> I expect we'll find few more similar disagreements and hopefully we'll resolve them :)
> 
> For me, this **is** practical. But for sure there is nothing we can go and implement/calculate using this document only.

I did not mean anything that you can directly implement, but rather some argument that it is at all feasible.

How about working through some examples?  I think it would improve readability as well (cf my point 2 in https://github.com/golemfactory/golem-architecture/pull/45#pullrequestreview-875158592)